### ISSUE TITLE
Feature Request: provide inline validation function in options

### ISSUE THREAD

**Issue by mwpastore** [offirgolan/ember-cp-validations#518]
If you have a really hairy inline validation function, it would be neat if you could pass it along in the options hash instead of as the first parameter. Something like this:

```javascript
foo: validator({
  validate(value, options, model, attribute) {
    // hairy monstrosity goes here
  }
})
```

Or, to maintain a consistent API:

```javascript
foo: validator('inline', {
  validate(value, options, model, attribute) {
    // ..
  }
})
```

**Comment by offirgolan** [comment]
I agree with you @mwpastore, this will definitely create a more uniform API. I would need to write a backwards compatible way of doing this which shouldnt be too hard but I just dont have the time right now. 

In terms of a `really hairy inline validation function`, I would suggest using a custom validator instead 😄

**Comment by mwpastore** [comment]
@offirgolan It looks like you implemented this in #569?

**Comment by offirgolan** [comment]
@mwpastore yup! Will be released with `v4.0.0` 😄
